### ISSUE TITLE
Bug: incomplete implementation of Files.newDirectoryStream(dir: Path, glob: String)

### ISSUE THREAD

**Issue by david-bouyssie** [scala-native/scala-native#1619]
After a private discussion on gitter with @LeeTibbert, we discovered that in the file https://github.com/scala-native/scala-native/blob/master/javalib/src/main/scala/java/nio/file/Files.scala the following method is missing a correct implementation:
```def newDirectoryStream(dir: Path, glob: String): DirectoryStream[Path]```

Here is the origin of the problem (see https://github.com/scala-native/scala-native/blob/master/javalib/src/main/scala/java/nio/file/PathMatcherImpl.scala):
```
object PathMatcherImpl {
  def apply(syntaxAndPattern: String): PathMatcher = {
    val colonIndex = syntaxAndPattern.indexOf(':')
    if (colonIndex == -1) {
      throw new IllegalArgumentException()
    }
    val syntax = syntaxAndPattern.substring(0, colonIndex)
    val pattern =
      syntaxAndPattern.substring(colonIndex + 1, syntaxAndPattern.length())

    if (syntax == "regex") new RegexPathMatcher(Pattern.compile(pattern))
    else throw new UnsupportedOperationException()
  }
}
```

The PathMatcherImpl doesn't support the glob pattern filter which can be defined in two ways:
* implicitly => ```Files.newDirectoryStream("./mydir/", "*.scala")```
* explicitly => ```Files.newDirectoryStream("./mydir/", "glob:*.scala")```

As a workaround I'm using this one line of code:
```new java.io.File("./mydir/").listFiles().filter(_.getName.endsWith(".scala"))```

**Comment by LeeTibbert** [comment]
I understand all to well why the original devo had to put a stake in the ground as to
what is supported.  To my thinking, an UnsupportedOperatationException is entirely
reasonable in the second/explicit case.

It is the first case, where the match method is not specified and defaults in hidden code
to glob: that seems to give SN an opportunity to improve.  I would have to check the Java
doc, but Java probably defaults to glob.

One possible better-than-nothing change might be to understand the Java default, then
change the SN default to regex and document it.   

I believe that regex is more powerful than glob and there are likely to be differences
in the match language, so there would be some rough edges. Perhaps  they are fewer and
less sharp than the existing ones?

It would be an interesting experiment if this ever percolates up the priority list.

**Comment by david-bouyssie** [comment]
To your wondering ```I would have to check the Java doc, but Java probably defaults to glob.``` I can say

**Comment by WojciechMazur** [comment]
Fixed in #2559
